src/autorest-core/lib/configuration.ts

```diff
@@ -44,6 +44,7 @@ export interface AutoRestConfigurationImpl {
   "package-version"?: string;
   "sync-methods"?: "all" | "essential" | "none";
   "payload-flattening-threshold"?: number;
+  "openapi-type"?: string
```

Comment:
This is to specify the swagger document type (ARM/Data-Plane/Default), this is the best I could think of.